app/upgrades/v18/upgrades.go

```diff
@@ -0,0 +1,216 @@
+package v17
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	recordskeeper "github.com/Stride-Labs/stride/v17/x/records/keeper"
+	recordtypes "github.com/Stride-Labs/stride/v17/x/records/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v17/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v17/x/stakeibc/types"
+)
+
+func NewIntFromString(val string) sdkmath.Int {
+	out_int, ok := sdkmath.NewIntFromString(val)
```

Comment:
I think the only reason is the INJ/EVMOS values overflow if they're regular ints, so string felt a bit easier. But no preference on my end on what casting method we use, as long as it can handle the big numbers!

Reply:
ah that makes sense!